hw/ip/edn/doc/_index.md

```diff
@@ -0,0 +1,270 @@
+Firmware must then wait until the current command is completed by polling the `REQ_MODE_SM_STS` bit in the {{< regref "SUM_STS" >}} register
+Once `REQ_MODE_SM_STS` reads zero, the state machine is idle and new firmware-driven commands can be passed to the CSRNG via the {{< regref "SW_CMD_REQ" >}} register.
+
+### Note on State Machine Shutdown Delays
+
+When leaving boot-time request mode or auto request mode, the EDN state machine waits for completion of the last command, before sending a shutdown acknowledgement to firmware.
+The longest possible commands are the `instantiate` or `reseed` requests, which typically take about 5ms, due to the time required to gather the necessary physical entropy.
+By contrast, the largest possible `generate` command allowed by [NIST SP 800-90A](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-90Ar1.pdf) is for 2<sup>19</sup> bits (or 4096 AES codewords).
+Assuming an AES encryption delay of 16 clocks, and a 100 MHz clock frequency, the longest allowable `generate` command would take only 0.7 ms to complete.
+
+### Note on Sharing of CSRNG Instance State Variables
+
+Once an application interface port has received an `instantiate` command, that application interface port then has access to a unique CSRNG instance, which is shared by all endpoints on the same EDN.
+Therefore from a security perspective, an attack to that particular CSRNG instance is an attack on all the endpoints that share the same EDN.
+Meanwhile, seeds and other state variables specific to a particular CSRNG instance are not shared between endpoints on *separate* EDN instances, or with any hardware devices with direct connections to dedicated CSRNG application interface ports.
+
+## Interactions with Peripheral Devices
+
+Peripheral ports distribute data to the endpoints using four signals: `req`, `ack`, `bus`, and `fips`.
+
+The EDN is activated in request/acknowledge mode whenever both `EDN_ENABLE` and `BCAST_MODE_DIS` bits are simultaneously asserted in {{< regref "CTRL" >}}.
+In request/acknowledge mode, fresh (i.e. previously unseen) random values are distributed to the endpoints via the parametrizable-width `bus` signal, in response to an `req` signal.
+Whenever new values are placed on the `bus`, the `ack` is asserted until the values are consumed by the endpoint, as indicated by simultaneous assertion of the `req` and `ack` signals in the same cycle.
+Otherwise `ack` is deasserted until enough fresh bits are received from CSRNG.
+
+The `fips` signal is used to identify whether the values received on the `bus`, have prepared with complete adherence to the recommendations in NIST SP 800-90.
+Typically if the `fips` signal is deasserted, it means the associated CSRNG instance has been instantiated with a pre-FIPS seed.
```

Comment:
Is there any other situation where the `fips` signal can be false?

Reply:
If we are removing broadcast mode, then no.
Will read: "If the `fips` signal is deasserted, it means the associated CSRNG instance has been instantiated with a pre-FIPS seed." in the next commit.